csrc/mmdeploy/codebase/mmcls/linear_cls.cpp

```diff
@@ -37,6 +37,7 @@ class LinearClsHead : public MMClassification {
     }
 
     auto class_num = (int)output.shape(1);
+    topk_ = topk_ < class_num ? topk_ : class_num;
```

Comment:
Modifying class member `top_k_` makes the module stateful, which is hard to reason locally.

Add `auto top_k = std::min(top_k_, class_num);` in `GetLabels` and use it as loop condition instead of modifying `top_k_`